test/jdk/java/awt/print/PrinterJob/PrintAllFonts.java

```diff
@@ -21,51 +21,59 @@
  * questions.
  */
 
-/**
- *
- * test
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.GraphicsEnvironment;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+
+/*
+ * @test
  * @bug 4884389 7183516
  * @key printer
+ * @library /test/lib /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary Font specified with face name loses style on printing
  * @run main/manual PrintAllFonts
  */
-
-import java.awt.*;
-import java.awt.print.*;
-import java.awt.GraphicsEnvironment;
-
 public class PrintAllFonts implements Printable {
 
-    static Font[] allFonts;
-    int fontNum = 0;
-    int startNum = 0;
-    int lineHeight = 18;
-    boolean done = false;
-    int thisPage = 0;
+    private static Font[] allFonts;
+    private int lineHeight = 18;
+    private int fontNum = 0;
+    private int startNum = 0;
+    private int thisPage = 0;
 
+    private static final String INSTRUCTIONS =
+            "You must have a printer available to perform this test.\n" +
+            "\n" +
+            "This bug is system dependent and is not always reproducible.\n" +
+            "A passing test will have all text printed with correct font style.";
```

Comment:
Clarify that the second column should be proper *italics*. I don't know how to explain it better; you can look at the PDFs attached to [JDK-4884389](https://bugs.openjdk.org/browse/JDK-4884389) to understand better what the bug was.

Reply:
I still suggest clarifying that the second column should always have the italic style and should use the italic font style if it's available (in the left column) instead of synthesised italic.

Look at the `1.4.2.pdf` file attached to [JDK-4884389](https://bugs.openjdk.org/browse/JDK-4884389).